src/main/java/com/splunk/logging/HttpEventCollectorSender.java

```diff
@@ -248,6 +304,48 @@ private String serializeEventInfo(HttpEventCollectorEventInfo eventInfo) {
         return event.toString();
     }
 
+    /**
+     * Parses the event message on the basis of message format.
+     * 
+     * @param message the event message string
+     * @param format the event message format
+     * 
+     * @return parsed event message based on format
+     */
+    private Object parseEventMessage(final String message, final MessageFormat messageFormat) {
+        // if message is null or blank then return without parsing
+        if (message == null || message.trim().length() == 0) {
+            return message;
+        }
+
+        switch (messageFormat) {
+            case JSON:
```

Comment:
Thanks for this nice suggestion @StShadow .

I have incorporated the changes. I would call the method as parse as it parses string into object representation. If it would have been other way around, then would have called it format.